### PR TITLE
An update

### PR DESCRIPTION
**Pull request by imolein**
* Formated copyright according to debians copyright-format
* Added discovery script
* Updated get script
* Create sudoers.d file instead of editing sudoers in postinst
* Removed old template
* Changed file paths to match paths in the other zabbix-check packages

### CODE REVIEW

**Comment by imolein** [comment]
You can review it, but don't merge it yet. I have to create and add a new template before.

File: usr/share/doc/zabbix-check-sshkey/zabbix_check_sshkey_template.xml [hunk 1]
```diff
@@ -1,78 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (istoph) [inline usr/share/doc/zabbix-check-sshkey/zabbix_check_sshkey_template.xml:1]: Why do we delete them definitively? Shouldn't we include a recent example?

File: usr/lib/zabbix-check/bin/zabbix-check-sshkey [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+fail() {
+    echo "ZBX_NOTSUPPORTED"
+    exit 1
+}
+
+
+[ -r "${authorized_keyfile}" ] || fail
+
+md5sum "${authorized_keyfile}" | cut -d" " -f1
```
Comment (istoph) [inline usr/lib/zabbix-check/bin/zabbix-check-sshkey]: We should consider switching to sha512sum.

File: usr/lib/zabbix-check/bin/zabbix-check-sshkey-discovery [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env sh
+
+set -e
+
+return_val="{\"data\":["
+
+# reading home paths of users from /etc/passwd and checks
+# if there is an authorized_keys file
+while read -r line; do
+    auth_keyfile_path="$(echo "${line}" | cut -d":" -f6)/.ssh/authorized_keys"
```
Comment (istoph) [inline usr/lib/zabbix-check/bin/zabbix-check-sshkey-discovery:10]: There may also be a `~/.ssh/authorized_keys2`, we should check that too. The file is not really well documented but here (https://www.debian.org/security/key-rollover/index.de.html) you can see that it is also used under Debian.

**Review by istoph — DISMISSED** [review]

File: usr/share/doc/zabbix-check-sshkey/zabbix_check_sshkey_template.xml (hunk 1 above)
Comment (imolein) [reply usr/share/doc/zabbix-check-sshkey/zabbix_check_sshkey_template.xml:1]: > You can review it, but don't merge it yet. I have to create and add a new template before.

Yes, I want to add a new one, that's why I tagged it as WIP and wrote this comment yesterday ;)

File: usr/lib/zabbix-check/bin/zabbix-check-sshkey (hunk 2 above)
Comment (imolein) [reply usr/lib/zabbix-check/bin/zabbix-check-sshkey]: done

File: usr/lib/zabbix-check/bin/zabbix-check-sshkey-discovery (hunk 3 above)
Comment (imolein) [reply usr/lib/zabbix-check/bin/zabbix-check-sshkey-discovery:10]: done

File: debian/zabbix-check-sshkey.examples [hunk 4]
```diff
@@ -0,0 +1 @@
+examples/zabbix-check-sshkey-template.xml
```
Comment (istoph) [inline debian/zabbix-check-sshkey.examples]: `dh_installexamples: error: Cannot find (any matches for) "examples/zabbix-check-sshkey-template.xml" (tried in .)`

**Comment by imolein** [comment]
Sooooo, I guess it's finished :)

File: debian/postrm [hunk 5]
```diff
@@ -18,24 +18,19 @@ set -e
 # the debian-policy package
 
 case "$1" in
-
     remove|purge)
-        grep -v 'zabbix-check-sshkey' /etc/sudoers > /etc/.sudoers
-        mv /etc/.sudoers /etc/sudoers
-        chmod 440 /etc/sudoers
```
Comment (istoph) [inline debian/postrm:25]: If you now take it out, it will not be removed when updating and will therefore appear twice.

Comment (imolein) [reply debian/postrm:25]: Oops, you are right. I thought this was some historical stuff and removed it without checking...

Comment (imolein) [reply debian/postrm:25]: Done

**Pull request merged**